research/berry/berrylib/fast_inla.py

```diff
@@ -217,30 +277,13 @@ def optimize_mode(self, y, n, fixed_arm_dim=None, fixed_arm_values=None):
 
         return theta_max, hess_inv
 
-    def log_joint(self, y, n, theta):
-        """
-        theta is expected to have shape (N, n_sigma2, n_arms)
-        """
-        theta_m0 = theta - self.mu_0
-        theta_adj = theta + self.logit_p1
-        exp_theta_adj = np.exp(theta_adj)
-        return (
-            0.5 * np.einsum("...i,...ij,...j", theta_m0, self.neg_precQ, theta_m0)
-            + self.logprecQdet
-            + np.sum(
-                theta_adj * y[:, None] - n[:, None] * np.log(exp_theta_adj + 1),
-                axis=-1,
-            )
-            + self.log_prior
-        )
-
-    def jax_inference(self, y, n):
+    def jax_inference(self, data):
         """
         See the numpy implementation for comments explaining the steps. The
         series of operations is almost identical in the JAX implementation.
         """
-        y = jnp.asarray(y)
-        n = jnp.asarray(n)
+        y = jnp.asarray(data[..., 0], dtype=np.float64)
```

Comment:
Why add float64 here?

Reply:
Oh, I can revert this. I was trying to see if the M1 issues we were having went away when I made very certain everything was float64. Those problems were solved with your PR though so we should be all set now. I'll remove this.